backend/compact-connect/tests/unit/test_api.py

```diff
@@ -0,0 +1,107 @@
+
+    @classmethod
+    def _when_testing_sandbox_stack_app(cls):
+        with open('cdk.json', 'r') as f:
+            context = json.load(f)['context']
+        with open('cdk.context.sandbox-example.json', 'r') as f:
+            context.update(json.load(f))
+
+        # Suppresses lambda bundling for tests
+        context['aws:cdk:bundling-stacks'] = []
+
+        app = CompactConnectApp(context=context)
+
+        return app
+
+    def _generate_expected_integration_object(self, handler_logical_id: str) -> dict:
+        return {
+            "Uri": {
+                "Fn::Join": [
+                    "",
+                    [
+                        "arn:aws:apigateway:us-east-1:lambda:path/2015-03-31/functions/",
+                        {
+                            "Fn::GetAtt": [handler_logical_id, "Arn"]
+                        },
+                        "/invocations"
+                    ]
+                ]
+            }
+        }
+
+    def test_synth_generates_provider_users_resource(self):
+        api_stack = self.app.sandbox_stage.api_stack
+        api_stack_template = Template.from_stack(api_stack)
+
+        # Ensure the resource is created with expected path
+        api_stack_template.has_resource_properties(
+            type=CfnResource.CFN_RESOURCE_TYPE_NAME,
+            props={
+                "ParentId": {
+                    # Verify the parent id matches the expected 'v1' resource
+                    'Ref': api_stack.get_logical_id(api_stack.api.v1_api.resource.node.default_child)
+                },
+                "PathPart": "provider-users"
+            })
+
+    def test_synth_generates_get_provider_users_me_endpoint_resource(self):
+        api_stack = self.app.sandbox_stage.api_stack
+        api_stack_template = Template.from_stack(api_stack)
+
+        # Ensure the resource is created with expected path
+        api_stack_template.has_resource_properties(
+            type=CfnResource.CFN_RESOURCE_TYPE_NAME,
+            props={
+                "ParentId": {
+                    # Verify the parent id matches the expected 'provider-users' resource
+                    'Ref': api_stack.get_logical_id(api_stack.api.v1_api.provider_users_resource.node.default_child)
+                },
+                "PathPart": "me"
+            })
+
+        # ensure the handler is created
+        api_stack_template.has_resource_properties(
+            type=CfnFunction.CFN_RESOURCE_TYPE_NAME,
+            props={
+                "Handler": "handlers.provider_users.get_provider_user_me"
+            })
+
+        # ensure the GET method is configured with the lambda integration and authorizer
+        api_stack_template.has_resource_properties(
+            type=CfnMethod.CFN_RESOURCE_TYPE_NAME,
+            props={
+                "HttpMethod": "GET",
+                # the provider users endpoints uses a separate authorizer from the staff endpoints
```

Comment:
do we not need to test this for the staff endpoints above?

Reply:
We do, I wasn't going to tackle that as part of this PR, since it is out of scope from what I was focusing on. The intent is that whenever work on any tickets related to the staff endpoints the developer should be adding test cases corresponding to their work. If you feel its in scope for me to add it as part of this ticket, I can take the time to add the test cases for it.

Reply:
Fine to leave out, but important enough to test that I just want to make sure it's being tracked somewhere, whether in a ticket or IA's internal notes.

Reply:
If we would like to track this, my recommendation would be to add it to the `Testing List` section of our PR template. This would make sense given that these tests should be added as part of feature request tickets, so anytime new cdk code is checked into a PR we can explicitly call out the requirement for CDK tests that cover API endpoint configuration. I'll update the template to reflect this.

Reply:
The commit has been added.

Reply:
That's great. I think my main concern - if I'm understanding correctly - is that we miss adding these tests for _existing_ endpoints / CDK code for the staff endpoints. Would that not be something we'd want to track outside of the PR template? (Since it's not associated with _new_ CDK code.)

Reply:
I think back-filling CDK tests for existing API endpoints is important to do (I really like Landon's approach here), though I'm not sure the importance for existing endpoints rises to the level that we need to create additional stories focused solely on back-filling. The existing endpoints have already been tested via manual process, so we know they work. There is a small regression risk for them that we can close by adding these tests but even that is limited because we will still have a manual verification process for code changes, which should catch something as big as a whole API method disappearing.

Really, this is a 'fail fast' improvement that allows devs to immediately see if they accidentally broke something more than it is a reliability improvement, since I think the risk of the lack of existing tests allowing a broken change all the way into production is still pretty small.

All that is to say, I think Landon and I should just make a point to add these tests for everything new and to back-fill relevant endpoints as we go, from now on.

Reply:
It's a matter of paying off tech debt. The tests should have been added when the API endpoints were initially created, but since they were not, that is now tech debt for this project. If management is fine with us creating tickets purely for tech debt without providing any additional features then yes, we could certainly create a ticket for that and track it in a sprint. I will bring this up in standup today, but if there is push back for this as not being a priority, I don't believe that should hold up this particular PR, since I have provided test coverage for the changes I have contributed.